packages/core/src/state/NetworkManager.ts

```diff
@@ -43,13 +43,16 @@ export default class NetworkManager implements Manager {
           switch (action.type) {
             case FETCH_TYPE:
               this.handleFetch(action, dispatch);
-              // This is the only case that causes any state change
-              // It's important to intercept other fetches as we don't want to trigger reducers during
-              // render - so we need to stop 'readonly' fetches which can be triggered in render
-              if (action.meta.optimisticResponse !== undefined) {
-                return next(action);
+              // Eliminate throttled fetches from future middlewares + reducer
+              // TODO: Maybe make throttling its own middleware?
+              if (action.meta.throttle && action.key in this.fetched) {
+                return Promise.resolve();
```

Comment:
why not make the function `async` so you don't need the explicit promise stuff?

Reply:
Targetting old versions using async/await transpilation added a ton of code, so I tried avoiding it. Even now it's kinda meh in this case since it would just be 'return'

Reply:
Yeah the generator transform of async await sucks; a promises one would be much better. Fair enough.